storage/store.go

```diff
@@ -564,6 +559,33 @@ func (s *Store) WaitForInit() {
 	s.initComplete.Wait()
 }
 
+func (s *Store) startUpdateGC() {
+	updateGC := func() {
+		minRCacheTS := s.ctx.Clock.Now().WallTime - GCResponseCacheExpiration.Nanoseconds()
+		// We don't actually GC Txn entries just yet. See #2062.
+		const minTxnTS = 0
+		// These timeouts are used each time an engine compaction is
+		// underway. Transaction records and response cache entries
+		// older than the respective timestamp are gc'ed.
+		s.engine.SetGCTimeouts(minTxnTS, minRCacheTS)
+	}
+
+	updateGC()
+	s.stopper.RunWorker(func() {
+		ticker := time.NewTicker(time.Minute)
```

Comment:
Maybe make this something like `GCResponseCacheExpiration / 10` so it can be more responsive with smaller values of `GCResponseCacheExpiration` (useful for testing at least)